File: java/metricsTenantUsage/src/main/java/com/ionic/samples/api/SampleConfig.java
package com.ionic.samples.api;

import java.io.FileReader;
import java.util.*;

import org.ini4j.Ini;

public class SampleConfig
{
    public static String ConfigFile = null;

    public static boolean Loaded = false;
    public static String AuthHeader = null;

    public static String APIURL = null;
    public static String TenantID = null;

    public static boolean LoadConfig()
    {
        Loaded = false;

        // Build path for common API sample config file
        //
        String userDir = System.getenv("USERPROFILE");
        ConfigFile = userDir + "/.ionicsecurity/IonicAPI.cfg";

        try {
            Ini ini = new Ini(new FileReader(ConfigFile));

            // Load tenant data
            //
            Ini.Section section = ini.get("Tenant");

            APIURL = section.get("URL");
            TenantID = section.get("ID");
            if (APIURL == null || APIURL.isEmpty() || TenantID == null || TenantID.isEmpty())
            {
                return Loaded;
            }

            // Load Auth data & build header string
            //
            section = ini.get("Authorization");

            String AuthType = section.get("Type");
            if (AuthType == null || AuthType.isEmpty())
            {
                AuthType = "Basic";
            }

            switch (AuthType)
            {
                case "Basic":
                    String user = section.get("User");
                    String pwd = section.get("Password");
                    if (user == null || user.isEmpty() || pwd == null || pwd.isEmpty())
                    {
                        return Loaded;
                    }

                    String toEncode = user + ":" + pwd;
                    String encodedString = Base64.getEncoder().encodeToString(toEncode.getBytes());
                    AuthHeader = "Basic " + encodedString;
                    break;

                case "Bearer":
                    String apiToken = section.get("APIToken");
                    if (apiToken == null || apiToken.isEmpty())
                    {
                        return Loaded;
                    }

                    AuthHeader = "Bearer " + apiToken;
                    break;
            }

        }
        catch (Exception ex) {
            return Loaded;
        }

        Loaded = true;
        return Loaded;
    }
}

File: csharp/scim-UserList/scim-UserList/SampleConfig.cs
using IniParser;
using IniParser.Model;
using System;
using System.IO;

namespace IonicAPISample
{
    public class SampleConfig
    {
        public static string ConfigFile { get; internal set; }
        public static bool Loaded { get; internal set; }

        public static string AuthHeader { get; internal set; }

        public static string APIURL { get; internal set; }
        public static string TenantID { get; internal set; }

        public static bool LoadConfig()
        {
            Loaded = false;

            // Build path for common API sample config file
            //
            String userDir = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
            ConfigFile = Path.Combine(userDir, ".ionicsecurity/IonicAPI.cfg");

            // Load tenant data
            //
            var parser = new FileIniDataParser();
            IniData configIni = parser.ReadFile(ConfigFile);

            APIURL = configIni["Tenant"]["URL"];
            TenantID = configIni["Tenant"]["ID"];
            if (String.IsNullOrEmpty(APIURL) || String.IsNullOrEmpty(TenantID))
            {
                return Loaded;
            }

            // Load Auth data & build header string
            //
            string AuthType = configIni["Authorization"]["Type"];
            if (String.IsNullOrEmpty(AuthType))     // Default to "Basic" if not specified
                AuthType = "Basic";

            switch (AuthType)
            {
                case "Basic":
                    string user = configIni["Authorization"]["User"];
                    string pwd = configIni["Authorization"]["Password"];
                    if (String.IsNullOrEmpty(user) || String.IsNullOrEmpty(pwd))
                    {
                        return Loaded;
                    }

                    var plainTextBytes = System.Text.Encoding.UTF8.GetBytes($"{user}:{pwd}");
                    AuthHeader = "Basic " + System.Convert.ToBase64String(plainTextBytes);
                    break;

                case "Bearer":
                    String apiToken = configIni["Authorization"]["APIToken"];
                    if (String.IsNullOrEmpty(apiToken))
                    {
                        return Loaded;
                    }

                    AuthHeader = $"Bearer {apiToken}";
                    break;
            }

            Loaded = true;
            return Loaded;
        }
    }
}
